src/GlobalPayments.Api/Utils/TypeLengthValue.cs

```diff
@@ -0,0 +1,81 @@
+﻿using GlobalPayments.Api.Entities;
+using GlobalPayments.Api.Terminals.Ingenico;
+using System;
+using System.Linq;
+using System.Text;
+
+namespace GlobalPayments.Api.Utils {
+    internal class TypeLengthValue {
+
+        private byte[] _data = new byte[0];
+        private TLVFormat _format = TLVFormat.Standard;
+
+        public TypeLengthValue() {
+
+        }
+
+        public TypeLengthValue(byte[] data) {
+            _data = data;
+        }
+
+        // Add TLV Format since Ingenico has different format of length when it comes to TLV Standard.
+        public TLVFormat TLVFormat {
+            get { return _format; }
+            set { _format = value; }
+        }
+
+
+        public object GetValue(byte type, Type returnType, TLVFormat? format = null) {
```

Comment:
Generics?

Not sure what this class is for really...

Reply:
I mean I know what TLV data is and that you're well... not really parsing it... but why?